### ISSUE TITLE
Dynamic Feature regression since 3.0.1 (loadKoinModules)

### ISSUE THREAD

**Issue by egorikftp** [InsertKoinIO/koin#1095]
**Describe the bug**
Hello, I have upgraded to 3.0.1 version and looks like Koin doesn’t work with the Dynamic Feature anymore.
Previously I loaded module during class creation like this:

```kotlin
class PostCreatorFragment : ScopeFragment(R.layout.fragment_post_creator) {
  
    init {
        loadKoinModules(postModule)
    }

    ...

    override fun onDestroy() {
        super.onDestroy()
        unloadKoinModules(postModule)
    }
}
```

Now, when I opening fragment I receive the following exception:

```kotlin
org.koin.core.error.NoScopeDefFoundException: No Scope Definition found for qualifer 'com.egoriku.ladyhappy.postcreator.presentation.PostCreatorFragment'
        at org.koin.core.registry.ScopeRegistry.createScope(ScopeRegistry.kt:126)
        at org.koin.core.Koin.createScope(Koin.kt:208)
        at org.koin.core.Koin.createScope$default(Koin.kt:204)
        at org.koin.androidx.scope.FragmentExtKt$fragmentScope$1.invoke(FragmentExt.kt:29)
        at org.koin.androidx.scope.FragmentExtKt$fragmentScope$1.invoke(Unknown Source:2)
        at org.koin.androidx.scope.LifecycleScopeDelegate.<init>(LifecycleScopeDelegate.kt:30)
        at org.koin.androidx.scope.LifecycleScopeDelegate.<init>(LifecycleScopeDelegate.kt:19)
        at org.koin.androidx.scope.FragmentExtKt.fragmentScope(FragmentExt.kt:28)
        at org.koin.androidx.scope.ScopeFragment.<init>(ScopeFragment.kt:38)
        at org.koin.androidx.scope.ScopeFragment.<init>(ScopeFragment.kt:35)
```

Looks like `_scopeDefinitions` don’t contain my scope from the dynamic feature and scope initiating before the `loadKoinModules` method

**Source code**: 
https://github.com/egorikftp/Lady-happy-Android/blob/e2fbec6c4f581b84e390caa4958e7f8e07c059e4/sources/dynamicFeature/postCreator/src/main/kotlin/com/egoriku/ladyhappy/postcreator/presentation/PostCreatorFragment.kt#L43

**Comment by arnaudgiuliani** [comment]
Should be fixed with new engine development - `c04433ede7ea13a55d4e385d9c8fc04ac03e5c95`

**Comment by egorikftp** [comment]
@arnaudgiuliani Issue reproduces in the latest version.

I have created the sample project with the crash: http://github.com/egorikftp/KoinBug

**Comment by egorikftp** [comment]
Yes, but it worked previously)

**Comment by arnaudgiuliani** [comment]
Ok, let me check your project 👍

**Comment by arnaudgiuliani** [comment]
With such new Scope API I need to relax Scope definition creation constraint. This is now fixed in b31bb07ab9b11431eff2150e6011542ab2f153e0
